sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1151,27 +1157,28 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
             requestRetryPolicy = new PartitionKeyMismatchRetryPolicy(collectionCache, requestRetryPolicy, collectionLink, options);
         }
         DocumentClientRetryPolicy finalRequestRetryPolicy = requestRetryPolicy;
-        if (document == null) {
-            throw new IllegalArgumentException("document");
-        }
-        if (StringUtils.isEmpty(documentLink)) {
-            throw new IllegalArgumentException("documentLink");
-        }
+        return ObservableHelper.inlineIfPossibleAsObs(() -> replaceDocumentInternal(documentLink, document, options, finalRequestRetryPolicy), requestRetryPolicy);
+    }
 
-        if (document == null) {
-            throw new IllegalArgumentException("document");
-        }
+    private Mono<ResourceResponse<Document>> replaceDocumentInternal(String documentLink, Object document,
+                                                                     RequestOptions options, DocumentClientRetryPolicy retryPolicyInstance) {
+        try {
+            if (StringUtils.isEmpty(documentLink)) {
+                throw new IllegalArgumentException("documentLink");
+            }
 
-        Document typedDocument = documentFromObject(document, mapper);
-        logger.debug("Replacing a Document. documentLink: [{}]", documentLink);
-        final String path = Utils.joinPath(documentLink, null);
-        final Map<String, String> requestHeaders = getRequestHeaders(options);
+            if (document == null) {
+                throw new IllegalArgumentException("document");
+            }
 
-        String content = toJsonString(typedDocument, mapper);
+            Document typedDocument = documentFromObject(document, mapper);
```

Comment:
We are not using Document here anymore as this will result in double serialization/desrializaiton.
replace with 
```
String content = toJsonString(typedDocument, mapper);	
```

Reply:
Discussed this offline, it was there before the retryContext pr, so keeping the same for now